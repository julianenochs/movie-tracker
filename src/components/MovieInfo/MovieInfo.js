import React from 'react';
import { connect } from 'react-redux';
import './MovieInfo.scss';
import { NavLink } from 'react-router-dom';

export const MovieInfo = props => {
  console.log('props', props);
  const selectMovie = props.selectMovieToDisplay(props.id);
  return (
    <section className='movie-info__card'>
      <h2 className='movie__title'>{props.selectMovie.title}</h2>
      <img
        src={`https://image.tmdb.org/t/p/original/${props.selectMovie.poster_path}`}
        alt='movie poster'
      />
      <p className='overview__info'>{props.selectMovie.overview}</p>
      <p>Release Date: {props.selectMovie.release_date}</p>
      <p>Popularity Voting Average: {props.selectMovie.vote_average}</p>
      <NavLink to='/' className='movies__button'>
        Back To Movies
      </NavLink>
    </section>
  );
};

const mapStateToProps = state => ({
  selectMovie: state.selectMovie
});

export default connect(
  mapStateToProps,
  null
)(MovieInfo);
