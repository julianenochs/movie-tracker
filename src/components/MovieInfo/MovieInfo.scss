
$button-color: #004263;
$button-shadow: #003149;

.movie-info__card {
  align-items: center;
  background-color: rgba(0, 31, 46, 0.78);
  border: 2px solid white;
  color: white;
  display: flex;
  flex-direction: column;
  margin: 3px;
  margin-block-start: 5px;
  height: fit-content;
  padding: 15px;
  width: 85%;
}

.movie__title {
  font-family: 'Staatliches', cursive;
  font-size: 1.7em;
  height: 8vh;
  letter-spacing: 3px;
}

.overview__info {
  font-size: 1.2em;
  width: 65%;
}

.movies__button {
  background-color: $button-color;
  border: none;
  box-shadow: 4px 4px $button-shadow;
  color: white;
  font-family: 'Barlow Condensed', sans-serif;
  font-size: 1em;
  height: 40px;
  letter-spacing: 5px;
  text-decoration: none;
  width: 120px;
}