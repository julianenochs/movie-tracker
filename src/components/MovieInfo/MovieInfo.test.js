import React from 'react';
import { shallow } from 'enzyme';
import { MovieInfo, mapStateToProps, mapDispatchToProps } from './MovieInfo';

describe('MovieInfo', () => {
  let wrapper;
  const mockSelectMovieToDisplay = jest.fn();
  const mockSelectMovie = [
    {
      title: 'Peerat Test Movie',
      release_date: '12/25/2019',
      vote_average: 10
    }
  ];

  it('should match the MovieInfo Snapshot', () => {
    wrapper = shallow(
      <MovieInfo
        selectMovie={mockSelectMovie}
        selectMovieToDisplay={mockSelectMovieToDisplay}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
