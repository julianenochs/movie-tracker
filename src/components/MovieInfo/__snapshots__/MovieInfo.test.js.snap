// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo should match the MovieInfo Snapshot 1`] = `ShallowWrapper {}`;
